### ISSUE TITLE
ViLT for Masked language Modelling is not working 

### ISSUE THREAD

**Issue by sanyalsunny111** [NielsRogge/Transformers-Tutorials#164]
@NielsRogge I am trying to use ViLT for MLM for pretraining (inspired by your inference tutorial) but the issue that I am facing is the ViltFeatureExtractor or the ViltProcessor.feature_extractor is giving me error (screenshot) when I am trying to send a batch of 32 images rather than just 1 image shown in your ViLT MLM inference tutorial.

`feature_extr = ViltFeatureExtractor.from_pretrained("dandelin/vilt-b32-mlm")
        encoding_pixels = feature_extr(image, image_mean=[0.48145466, 0.4578275, 0.40821073],
                                       image_std=[0.26862954, 0.26130258, 0.27577711],
                                       return_tensors='pt').pixel_values`

`encoding_pixels = batched_featurextr(image, processor=processor).to(device)
encoding_pixels = processor.feature_extractor(image, image_mean=[0.48145466, 0.4578275, 0.40821073],
                                                      image_std=[0.26862954, 0.26130258, 0.27577711],
                                                      return_tensors='pt').pixel_values`

![image](https://user-images.githubusercontent.com/36811567/185009431-4a754a8b-28d7-450c-a4fd-9a7198ffccca.png)

**Comment by sanyalsunny111** [comment]
@NielsRogge a second issue could you please let me know why you have just used two inputs to the model i.e. input_ids (understandable as you are masking it for MLM) and pixel_values and why not pixel_masks, attention_mask, token_type_ids etc? Could you please expalin a bit?

**Comment by NielsRogge** [comment]
Hi,

The issue may be caused by the fact that the images aren't RGB (you need to call `.convert('RGB')` on the images before providing them to the feature extractor). 

It would be very cool to recreate this demo: https://github.com/dandelin/ViLT#out-of-the-box-mlm--visualization-demo but with the Huggingface model, and host the demo on Huggingface Spaces.

**Comment by sanyalsunny111** [comment]
> Hi,
> 
> The issue may be caused by the fact that the images aren't RGB (you need to call `.convert('RGB')` on the images before providing them to the feature extractor).
> 
> It would be very cool to recreate this demo: https://github.com/dandelin/ViLT#out-of-the-box-mlm--visualization-demo but with the Huggingface model, and host the demo on Huggingface Spaces.

I have converted the images to RGB, let me provide a but more details on this when I am trying to feature extract images one by one with a for loop like code below I am not getting any error. But when I am trying to send a batch the error mentioned in my first post is happening. Do you think it is something related to PIL library ?
`def batched_featurextr(batch_image, processor):
    encodings_ = list()
    for j in range(len(batch_image)):
        img_ = batch_image[j].cpu()
        # img_ = Image.fromarray((255 * image[j].permute(1, 2, 0)).numpy().astype(np.uint8))
        encoding_pixels = processor.feature_extractor(img_, image_mean=[0.48145466, 0.4578275, 0.40821073],
                                                      image_std=[0.26862954, 0.26130258, 0.27577711],
                                                      return_tensors="pt").pixel_values
        # encoding_pixels = processor(img_, image_mean=[0.48145466, 0.4578275, 0.40821073],
        #                             image_std=[0.26862954, 0.26130258, 0.27577711],
        #                             return_tensors='pt').pixel_values
        # encoding_pixelpad = processor.feature_extractor.pad_and_create_pixel_mask(encodings_pixels, return_tensors="pt")
        # encodings_.append(transformers.ViltFeatureExtractor(img_, do_resize=False, do_normalize=False, return_tensors='pt').pixel_values)
        encodings_.append(encoding_pixels)
    # encoding = torch.cat(encodings_, dim=0)
    encoding = torch.cat(encodings_)
    return encoding`
    
    P.S: Sure if my experiment succeeds I will have a huggingface demo and host it. But please help get there.

**Comment by sanyalsunny111** [comment]
This issue has been resolved. For future reference so, the solution here is if you wish to pass a batch of images to ViLTprocessor or ViLT Feature extractor pass it as a list, sample code:
`encoding_pixels = feature_extr(list(image), image_mean=[0.48145466, 0.4578275, 0.40821073],
                                       image_std=[0.26`

**Comment by NielsRogge** [comment]
Hi, not sure why you're passing the `image_mean` and `image_std` to the feature extractor when calling it. Those can be provided at initialization.

**Comment by sanyalsunny111** [comment]
True that was a mistake thank you for highlighting: The below code is the corrected version. But the basic issue has been resolved which is if you wish to send a batch of images through ViLT processor make them a list. I hope this is true for all other Feature Extracters as well.
`feature_extr = ViltFeatureExtractor.from_pretrained("dandelin/vilt-b32-mlm", do_resize=False, do_normalize=True,
                                                            image_mean=[0.48145466, 0.4578275, 0.40821073],
                                                            image_std=[0.26862954, 0.26130258, 0.27577711])
        encoding_pixels = feature_extr(list(image), return_tensors='pt').pixel_values`
